server/src/main/java/org/elasticsearch/search/fetch/subphase/highlight/HighlightUtils.java

```diff
@@ -46,24 +43,10 @@ private HighlightUtils() {
      * Load field values for highlighting.
      */
     public static List<Object> loadFieldValues(MappedFieldType fieldType,
-                                               FetchSubPhase.HitContext hitContext,
-                                               boolean forceSource) throws IOException {
-        //percolator needs to always load from source, thus it sets the global force source to true
-        List<Object> textsToHighlight;
-        if (forceSource == false && fieldType.isStored()) {
```

Comment:
Dropping this is a behavior change - this behavior of defaulting to stored fields was something folks were really keen on historically.

Reply:
Is this something that we can build into the ValueFetchers themselves? If it's a performance enhancement here, then it presumably will be elsewhere as well?

Reply:
> Is this something that we can build into the ValueFetchers themselves? If it's a performance enhancement here, then it presumably will be elsewhere as well?

It'd be *super* nice to have one way to think about this. So, hopefully!

Reply:
+1 to explore automatically loading stored fields when they're available. Beyond highlighting, this would simplify some field retrieval workflows where users currently need to request both `_source` and `stored_fields`. This seems like it'd be a bigger change, maybe it'd make sense as a follow-up issue or PR ?

Reply:
I think that using stored fields if they exist is kind of a blocker for this change, or, at least, for the part of this change that drops the stored fields stuff.

Reply:
Yes, I think we want to make that change first.  The tricky part will be communicating to the SourceLookup (or whatever stored-fields equivalent we come up with) which fields we want to load up-front.  I wonder what the cost of just automatically reading all stored fields in the Visitor would be?

Reply:
Could we just restore the conditional `if (forceSource == false && fieldType.isStored()) { ... ` here in order not to block this change on a larger one around stored fields ?

Reply:
I think that means we need to add back in the 'ignoreAbove' logic, which was the main reason I tried this out.

Reply:
I don't think so, because stored fields already respect the `ignore_above` option for keywords, so they'll be naturally excluded.

Reply:
That was the crux that I was missing, thank you!  Have pushed an update